### PR TITLE
DEPS: Drop Python 3.9

### PR DESCRIPTION
**Pull request by mroeschke**
Dropping Python 3.9 support per [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table) and [SPEC-0](https://scientific-python.org/specs/spec-0000/#support-window)

### CODE REVIEW

File: scripts/tests/data/deps_minimum.toml [hunk 1]
```diff
@@ -40,7 +40,6 @@ classifiers = [
     'Programming Language :: Python :: 3',
     'Programming Language :: Python :: 3 :: Only',
     'Programming Language :: Python :: 3.8',
```
Comment (tqa236) [inline scripts/tests/data/deps_minimum.toml:42]: Should Python :: 3.8 be removed as well?

Comment (mroeschke) [reply scripts/tests/data/deps_minimum.toml:42]: Good catch. Yes definitely

Comment (mroeschke) [reply scripts/tests/data/deps_minimum.toml:42]: (This is a testing file so not critical but still useful)

**Comment by twoertwein** [comment]
I think there are a few `sys.version_info` left that compare aginst 3.10, for example, in [pandas._typing](https://github.com/pandas-dev/pandas/blob/b9bfc0134455d8a5f269366f87d2a8567aa4269c/pandas/_typing.py#L95C8-L95C24). Pyupgrades's `--py310-plus` might take care of that automatically (currently set to 3.9)

File: .github/workflows/package-checks.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        python-version: ['3.9', '3.10', '3.11']
+        python-version: ['3.10', '3.11']
```
Comment (twoertwein) [inline .github/workflows/package-checks.yml:56]: Should 3.12 be in that list?

(`test_with_missing_lzma_runtime` is failing)

File: pandas/_typing.py [hunk 3]
```diff
@@ -90,18 +90,12 @@
...
-            Concatenate,
-            ParamSpec,
-            TypeGuard,
-        )
+    from typing import (
+        ParamSpec,
+        SupportsIndex,
+    )
+    from typing import Concatenate  # pyright: ignore[reportUnusedImport]
+    from typing import TypeGuard  # pyright: ignore[reportUnusedImport]
```
Comment (Dr-Irv) [inline pandas/_typing.py:98]: I think you can merge these into one statement, e.g.
```python
    from typing import (
        Concatenate, # pyright: ignore[reportUnusedImport]
        ParamSpec,
        SupportsIndex,
        TypeGuard, # pyright: ignore[reportUnusedImport]
    )
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
I only looked at `_typing.py`

File: pandas/_typing.py (hunk 3 above)
Comment (mroeschke) [reply pandas/_typing.py:98]: Looks like isort is forcing these imports to be separate. Is that OK with you?

File: .github/workflows/package-checks.yml (hunk 2 above)
Comment (mroeschke) [reply .github/workflows/package-checks.yml:56]: Good call. Will add

Comment (mroeschke) [reply .github/workflows/package-checks.yml:56]: Looks like `conda-verify` doesn't support Python 3.12 yet so we can't add this yet

**Review by twoertwein — APPROVED** [review]

File: pandas/_typing.py (hunk 3 above)
Comment (Dr-Irv) [reply pandas/_typing.py:98]: That's fine.

**Review by WillAyd — APPROVED** [review]
lgtm. Do we want to wait until we get closer to 3.0 release to merge?

**Comment by mroeschke** [comment]
> Do we want to wait until we get closer to 3.0 release to merge?

I guess we could merge anytime, but I would wait a little just in case people are still using 3.9 nightlies

**Comment by mroeschke** [comment]
Since it seemed like there was approval a few months ago, I'll merge at the end of the week if there are no objections

**Comment by WillAyd** [comment]
Works for me

**Pull request merged**